### DISCUSSION TITLE
Does Polly chaos engineering work with Kafka?

### DISCUSSION THREAD

**Discussion by 01vadim10** [App-vNext/Polly#2082] [Q&A] [answered]
Can I inject chaos for Kafka to test my app resilience for producers and consumers logic using Polly?

**Comment by peter-csala** [comment, marked as answer]
With the [Behavior chaos](https://www.pollydocs.org/chaos/behavior.html) you can write any arbitrary logic to setup the scene for your chaos experiment. You just have to write yourself the poison pill message publishing logic and pass that to the `BehaviorGenerator`.

**Reply by 01vadim10** [reply]
Thank you, @peter-csala !
